internal/cmd/envs_test.go

```diff
@@ -45,3 +56,151 @@ func Test_envs_ls_by_provider(t *testing.T) {
+
+	// Image unset
+	res = execute(t, nil, "envs", "create", "test-env")
+	res.error(t)
+	res.stderrContains(t, "fatal: required flag(s) \"image\" not set")
+
+	// Image not imported
+	res = execute(t, nil, "envs", "create", "test-env", "--image=doestexist")
+	res.error(t)
+	res.stderrContains(t, "fatal: image not found - did you forget to import this image?")
+
+	ensureImageImported(ctx, t, testCoderClient, "ubuntu")
+
+	name := randString(10)
+	cpu := 2.3
+
+	// attempt to remove the environment on cleanup
+	t.Cleanup(func() { _ = execute(t, nil, "envs", "rm", name, "--force") })
+
+	res = execute(t, nil, "envs", "create", name, "--image=ubuntu", fmt.Sprintf("--cpu=%f", cpu))
+	res.success(t)
+
+	res = execute(t, nil, "envs", "ls")
+	res.success(t)
+	res.stdoutContains(t, name)
+
+	var envs []coder.Environment
+	res = execute(t, nil, "envs", "ls", "--output=json")
+	res.success(t)
+	res.stdoutUnmarshals(t, &envs)
+	env := assertEnv(t, name, envs)
+	assert.Equal(t, "env cpu", cpu, float64(env.CPUCores), floatComparer)
+
+	res = execute(t, nil, "envs", "watch-build", name)
+	res.success(t)
+
+	// edit the CPU of the environment
+	cpu = 2.1
+	res = execute(t, nil, "envs", "edit", name, fmt.Sprintf("--cpu=%f", cpu), "--follow")
+	res.success(t)
+
+	// assert that the CPU actually did change after edit
+	res = execute(t, nil, "envs", "ls", "--output=json")
+	res.success(t)
+	res.stdoutUnmarshals(t, &envs)
+	env = assertEnv(t, name, envs)
+	assert.Equal(t, "env cpu", cpu, float64(env.CPUCores), floatComparer)
+
+	res = execute(t, nil, "envs", "rm", name, "--force")
+	res.success(t)
+}
+
+func assertEnv(t *testing.T, name string, envs []coder.Environment) *coder.Environment {
+	for _, e := range envs {
+		if name == e.Name {
+			return &e
+		}
+	}
+	slogtest.Fatal(t, "env not found", slog.F("name", name), slog.F("envs", envs))
+	return nil
+}
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func randString(length int) string {
+	const charset = "abcdefghijklmnopqrstuvwxyz"
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+var floatComparer = cmp.Comparer(func(x, y float64) bool {
+	delta := math.Abs(x - y)
+	mean := math.Abs(x+y) / 2.0
+	return delta/mean < 0.001
+})
+
+// this is a stopgap until we have support for a `coder images` subcommand
+// until then, we want can use the coder.Client to ensure our integration tests
+// work on fresh deployments.
+func ensureImageImported(ctx context.Context, t *testing.T, client coder.Client, img string) {
+	orgs, err := client.Organizations(ctx)
+	assert.Success(t, "get orgs", err)
+
+	var org *coder.Organization
+search:
+	for _, o := range orgs {
+		for _, m := range o.Members {
+			if m.Email == os.Getenv("CODER_EMAIL") {
+				o := o
+				org = &o
+				break search
+			}
+		}
+	}
+	if org == nil {
+		slogtest.Fatal(t, "failed to find org of current user")
+		return // help the linter out a bit
```

Comment:
Seems strange to have a return after a fatal but I guess there's something going on with the linter that's not obvious to me. The comment doesn't  help explain why so I think we can just elaborate on the comment or just add a `nolint`.  Either way no big deal just my general thoughts reading.